client/app/src/main/java/com/microsoft/research/karya/ui/registration/RegistrationViewModel.kt

```diff
@@ -0,0 +1,178 @@
+package com.microsoft.research.karya.ui.registration
+
+import android.util.Log
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.microsoft.research.karya.R
+import com.microsoft.research.karya.data.exceptions.IncorrectAccessCodeException
+import com.microsoft.research.karya.data.exceptions.IncorrectOtpException
+import com.microsoft.research.karya.data.exceptions.PhoneNumberAlreadyUsedException
+import com.microsoft.research.karya.data.exceptions.UnknownException
+import com.microsoft.research.karya.data.repo.KaryaFileRepository
+import com.microsoft.research.karya.data.repo.LanguageRepository
+import com.microsoft.research.karya.data.repo.WorkerRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import java.lang.Exception
+import javax.inject.Inject
+import kotlin.properties.Delegates
+import kotlin.reflect.typeOf
+
+// TODO: ADD dependencies in the constructor
+@HiltViewModel
+class RegistrationViewModel @Inject constructor(
+    private val workerRepository: WorkerRepository,
+    private val languageRepository: LanguageRepository,
+    private val karyaFileRepository: KaryaFileRepository,
+) : ViewModel() {
+
+    enum class OtpVerifyState {
+        NOT_ENTERED,
+        SUCCESS,
+        FAIL
+    }
+
+    enum class OtpSendState {
+        NOT_SENT,
+        SUCCESS,
+        FAIL
+    }
+
+    private val _openDashBoardFromOTP = MutableLiveData<Boolean>()
+    val openDashBoardFromOTP: LiveData<Boolean>
+        get() =_openDashBoardFromOTP
+
+    private val _openProfilePictureFragmentFromOTP = MutableLiveData<Boolean>()
+    val openProfilePictureFragmentFromOTP: LiveData<Boolean>
+        get() = _openProfilePictureFragmentFromOTP
+
+    private val _currOtpVerifyState = MutableLiveData<OtpVerifyState>()
+    val currOtpVerifyState: LiveData<OtpVerifyState>
+        get() = _currOtpVerifyState
+
+    private val _currOtpSendState = MutableLiveData<OtpSendState>()
+    val currOtpSendState: LiveData<OtpSendState>
+        get() = _currOtpSendState
+
+    private val _currOtpResendState = MutableLiveData<OtpSendState>()
+    val currOtpResendState: LiveData<OtpSendState>
+        get() = _currOtpResendState
+
+    var phoneNumberFragmentErrorId by Delegates.notNull<Int>()
+    var otpFragmentErrorId by Delegates.notNull<Int>()
+
+    init {
+        _openDashBoardFromOTP.value = false
+        _openProfilePictureFragmentFromOTP.value = false
+
+        _currOtpVerifyState.value = OtpVerifyState.NOT_ENTERED
+        _currOtpSendState.value = OtpSendState.NOT_SENT
+        _currOtpResendState.value = OtpSendState.NOT_SENT
+    }
```

Comment:
Please combine declaration and initialization together.

e.g, `private val _openDashBoardFromOTP = MutableLiveData<Boolean>(false)`